### PR TITLE
Use firebase for documentation deployments.

### PR DESCRIPTION
**Pull request by haroldsphinx**
- firebase deploy

### CODE REVIEW

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by Martin-Kiss — APPROVED** [review]

File: .github/workflows/docs-firebase-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Firebase Deploy Gcloud
```
Comment (KaiserKarel) [inline .github/workflows/docs-firebase-deploy.yml:1]:
```suggestion
name: Deploy Documentation
```

File: .github/workflows/docs-firebase-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        run: |
+         rustup set profile minimal
+         rustup toolchain install stable
+         rustup default stable
+
+      - name: Install mdbook
+        run: |
+          mkdir bin
+          curl -sSL https://github.com/rust-lang/mdBook/releases/download/v0.4.14/mdbook-v0.4.14-x86_64-unknown-linux-gnu.tar.gz | tar -xz --directory=bin
+          echo "$(pwd)/bin" >> $GITHUB_PATH
```
Comment (KaiserKarel) [inline .github/workflows/docs-firebase-deploy.yml:32]: We have/should have a nix step for this; we never manually install rust in CI, as that will lead to different versions.

**Review by cor — CHANGES_REQUESTED** [review]
Just run `nix build ".#composable-book"` instead of installing all of this manual tooling

**Comment by cor** [comment]
Not using Nix. Closing this one in favor of including it in the next Docusaurus PR

**Pull request closed without merging**